buzzmobile/tests/sense/test_inputer.py

```diff
@@ -1,55 +1,21 @@
-import rostest
 import unittest
-import rospy
-import time
 from std_msgs.msg import String
-import roslaunch
+from tests import rostest_utils
 
-PKG = 'buzzmobile'
 NAME = 'test_inputer'
 
 
+@rostest_utils.with_roscore
 class TestInputer(unittest.TestCase):
 
-    def run_node_with_callback(self, callback):
-        rospy.init_node(NAME, anonymous=True)
-
-        rospy.Subscriber('destination', String, callback)
-        timeout = time.time() + 2.0
-
-        while time.time() < timeout and not rospy.is_shutdown():
-            time.sleep(0.1)
-
-    def __init__(self, *args):
-        super().__init__(*args)
-
-
     def setUp(self):
         self.success = False
-        node = roslaunch.core.Node('buzzmobile', 'inputer.py')
-        launch = roslaunch.scriptapi.ROSLaunch()
-        launch.start()
-        self.process = launch.launch(node)
-
-    def test_inputer_no_input(self):
-        def cb(data):
-            self.success = (data.data == '')
-
-        self.run_node_with_callback(cb)
-        assert self.success
 
+    @rostest_utils.launch_node('buzzmobile', 'inputer.py', 'params.launch')
```

Comment:
Oh that's interesting. What you're doing in the background is adding the node to the launch file and launching everything together, right?

In answer to your question, I think that might be preferable. My main concern in separating things is if we want to launch multiple nodes for an integration test, there is currently no easy way of both using a launch file (say, for rosparams), and starting up multiple nodes.

So either way would work as an api:

```python
@launch_nodes(('buzzmobile', 'gps_mapper'), ('buzzmobile', 'lidar_to_frame'), ('buzzmobile', 'frame_merger'), launch='params.launch')
def test_merging_frames_integration(self):
    def cb(world_model):
        self.success = world_model is correct
    with check_topic('world_model', Image, cb):
        sleep(1) # or something, it'll take time to merge. not sure what's best practice
        assert self.success
```

or

```python
@with_launch_file('params.launch')
@launch_node('buzzmobile', 'gps_mapper')
@launch_node('buzzmobile', 'lidar_to_frame')
@launch_node('buzzmobile', 'frame_merger')
def test_merging_frames_integration(self):
    def cb(world_model):
        self.success = world_model is correct
    with check_topic('world_model', Image, cb):
        sleep(1) # or something, it'll take time to merge. not sure what's best practice
        assert self.success
```

Note that in the second one, `@with_launch_file` is also optional. So running nodes without a launch file is possible